arch/lkl/kernel/init.c

```diff
@@ -6,6 +6,25 @@ int lkl_init(struct lkl_host_operations *ops)
 {
 	lkl_ops = ops;
 
+#ifdef CONFIG_LKL_HOST_MEMCPY
+	if (!ops->memcpy) {
+		lkl_printf("unexpected NULL lkl_host_ops member: memcpy\n");
+		return -1;
+	}
+#endif
+#ifdef CONFIG_LKL_HOST_MEMSET
+	if (!ops->memset) {
+		lkl_printf("unexpected NULL lkl_host_ops member: memset\n");
+		return -1;
+	}
+#endif
+#ifdef CONFIG_LKL_HOST_MEMMOVE
+	if (!ops->memmove) {
+		lkl_printf("unexpected NULL lkl_host_ops member: memmove\n");
+		return -1;
+	}
+#endif
+
```

Comment:
coding-style.rst encourages avoiding `#ifdef CONFIG_...` outside of header files. How about something like:
```
        if ((IS_ENABLED(CONFIG_LKL_HOST_MEMCPY) && !ops->memcpy)
         || (IS_ENABLED(CONFIG_LKL_HOST_MEMSET) && !ops->memset)
         || (IS_ENABLED(CONFIG_LKL_HOST_MEMMOVE) && !ops->memmove)) {
                lkl_printf("unexpected NULL lkl_host_ops memory I/O call\n");
                return -1;
        }
```
The compiler should still be able to optimize it out for non-posix hosts.

Reply:
Done, thanks!